lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -146,6 +146,144 @@ def formatted_message_and_backtrace(colorizer, indentation)
 
           formatted
         end
+
+        # @private
+        # Configuring the `ExceptionPresenter` with the right set of options to handle
+        # pending vs failed vs skipped and aggregated (or not) failures is not simple.
+        # This clas takes care of building an appropriate `ExceptionPresenter` for the
+        # provided example.
+        class Factory
+          def build
+            ExceptionPresenter.new(@exception, @example, options)
+          end
+
+        private
+
+          def initialize(example)
+            @example          = example
+            @execution_result = example.execution_result
+            @exception        = if @execution_result.status == :pending
+                                  @execution_result.pending_exception
+                                else
+                                  @execution_result.exception
+                                end
+          end
+
+          def options
+            with_multiple_error_options_as_needed(@exception, pending_options || {})
+          end
+
+          def pending_options
+            if @execution_result.pending_fixed?
+              {
+                :description_formatter => Proc.new { "#{@example.full_description} FIXED" },
+                :message_color         => RSpec.configuration.fixed_color,
+                :failure_lines         => [
+                  "Expected pending '#{@execution_result.pending_message}' to fail. No Error was raised."
+                ]
+              }
+            elsif @execution_result.status == :pending
+              {
+                :message_color    => RSpec.configuration.pending_color,
+                :detail_formatter => PENDING_DETAIL_FORMATTER
+              }
+            end
+          end
+
+          def with_multiple_error_options_as_needed(exception, options)
```

Comment:
:shrug: (why isn't there an emoji for that?)

Probably.  I've spent so much time refactoring the exception presentation logic to get it to this point that I'm kinda out of  ideas.  There was already a method like this before this PR -- I just moved it and renamed it in this PR.

If you've got a concrete for an improvement here, please open a follow up PR after this one is merged :).